source/plugin/basics/blocks/tileentities.rst

```diff
@@ -0,0 +1,84 @@
+==========================
+Working with Tile Entities
+==========================
+
+Tile entities are blocks that are capable of additional operations like autonomically crafting (furnaces and brewing stands) or provide effects (like a beacon or a note block). They also contain additional data like an inventory or text (like chests, signs or command blocks).
+
+Identifying Tile Entities and their Type
+========================================
+
+Again, it all starts with a ``Location``. The ``getTileEntity()`` function will return the tile entity corresponding to the block or ``Optional.absent()`` if the block is not a tile entity.
+
+ .. code-block:: java
+ 
+	public boolean isTileEntity(Location block) {
+		return block.getTileEntity().isPresent();
+	}
+	
+The type of a tile entity can then be obtained by the ``getType()`` function which returns a ``TileEntityType``. Which can then be compared similar to a ``BlockType``. After performing this check the ``TileEntity`` variable can safely be cast to the according subtype.
+
+ .. code-block:: java
+ 
+	public boolean isJukebox(TileEntity entity) {
+		return entity.getType() == TileEntityTypes.JUKEBOX;
+	}
+	
+	public Optional<Jukebox> getJukebox(TileEntity entity) {
+		if (isJukebox(entity)) {
+			return Optional.of((Jukebox) entity);
+		} else {
+			return Optional.absent();
+		}
+	}
```

Comment:
You pinpoint exactly what I felt when writing those examples. I did it multiple times that I returned an `Optional` instead of writing much simpler code because I felt it to be more in line with the Sponge coding style.
I guess I'll just go with your second suggestion and act like I stole the relevant lines from bigger functions.